### PR TITLE
Rough draft for lfs hero app

### PR DESCRIPTION
**Pull request by danielRicaud**

### CODE REVIEW

File: examples/opentdf-lfs-hero-app/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# README
```
Comment (dmihalcik-virtru) [inline examples/opentdf-lfs-hero-app/README.md:1]:
```suggestion
# Large File Sharing
```
What does LFS stand for?

File: examples/opentdf-lfs-hero-app/package.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "proxy": "http://localhost:3001",
+  "dependencies": {
+    "@ant-design/icons": "^4.7.0",
+    "@opentdf/client": "file:opentdf-client-0.2.1.tgz",
+    "@react-keycloak/web": "^3.4.0",
+    "antd": "^4.20.5",
+    "copyfiles": "^2.4.1",
```
Comment (dmihalcik-virtru) [inline examples/opentdf-lfs-hero-app/package.json:20]:
```suggestion
```
Is this used?

File: examples/opentdf-lfs-hero-app/src/utils/index.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export const validateJson = (jsonString) => {
```
Comment (dmihalcik-virtru) [inline examples/opentdf-lfs-hero-app/src/utils/index.js:1]: This seems to be a duplicate function that is also declared above

File: examples/opentdf-lfs-hero-app/README.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+- Once all services are up and running open the `Keycloak` console at `http://localhost:65432/auth` to configure the browser client to support this app
+
+   - Username: `keycloakadmin`
+   - Password: `mykeycloakpassword`
+
+- Navigate to the `abacus-web` client in the `tdf` realm of `Keycloak` and add `http://localhost:3001/*` to the list of `Valid Redirect URIs`. Be sure to save your changes at the bottom of the form.
+![Valid Redirect URI added to client](./resource/redirectUrl.png)
+
+- Navigate to the `Mappers` tab of the `abacus-web` client in the realm `tdf`. Here we'll add the appropriate mapper for the client to be able to apply claims
```
Comment (dmihalcik-virtru) [inline examples/opentdf-lfs-hero-app/README.md:18]: File a bug to fix bootstrap if it needs fixing. Don't let annoyances like this slide

File: examples/opentdf-lfs-hero-app/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# README
+
+### Purpose
+
+- A practical React based example of OIDC authentication against Keycloak using OpenTDF's [client-web](https://github.com/opentdf/client-web) SDK to send encrypted TDF streams, or download/decrypt TDF streams, to and from an S3 compatible remote store
```
Comment (dmihalcik-virtru) [inline examples/opentdf-lfs-hero-app/README.md:5]: More description here would be nice.

- walkthrough how to use the app
- details about how to configure S3 for people who aren't used to it

An alternative might be making an in depth walkthrough in YouTube with iMovie or similar

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]
Can we move the folder to `examples/remote-files` or `example/large-files`? opentd is redundant, hero-app is a code name, and LFS Is obscure

File: examples/opentdf-lfs-hero-app/src/App.js [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+            <Button onClick={() => lfsDownload(text, record, index)}>Download/Decrypt</Button>
+          </Spin>
+        </div>
+      ),
+    }
+  ];
+
+  keycloak.onAuthError = console.log;
+
+  const validateJsonStr = (jsonString) => {
```
[the remark this answers is not in the document]
Comment (danielRicaud) [reply examples/opentdf-lfs-hero-app/src/App.js:71]: Looks like S3 will bubble up an error through our SDK that's shown in the dev tools! Here it's complaining about missing `Bucket` after I misspelled it `Buckett`
image

File: examples/opentdf-lfs-hero-app/README.md (hunk 4 above)
Comment (danielRicaud) [reply examples/opentdf-lfs-hero-app/README.md:18]: I'll add this as a definition of done for the infrastructure `tilt up` ticket

**Review by cassandrabailey293 — DISMISSED** [review]

**Review by cassandrabailey293 — APPROVED** [review]

**Pull request merged**